src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.orgrolemapping.data;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import uk.gov.hmcts.reform.orgrolemapping.controller.BaseTestIntegration;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ProfileRefreshQueueRepositoryIntegrationTest extends BaseTestIntegration {
+
+    @Autowired
+    private ProfileRefreshQueueRepository profileRefreshQueueRepository;
+
+    @Test
+    public void shouldInsertOrganisationProfileIds() {
+        List<String> organisationProfileIds = List.of("SOLICITOR_ORG", "DWP_GOV_ORG", "HMRC_GOV_ORG");
+        String orgProfileIds = String.join(",", organisationProfileIds);
+
+        profileRefreshQueueRepository.upsertOrganisationProfileIds(orgProfileIds, 1);
+
+        List<ProfileRefreshQueueEntity> profileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(profileRefreshQueueEntities, 3, 1);
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {
+        "classpath:sql/insert_organisation_profiles.sql"
+    })
+    public void shouldHandleDuplicateOrganisationProfileIds() {
+        List<ProfileRefreshQueueEntity> profileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(profileRefreshQueueEntities, 1, 1);
+
+        List<String> organisationProfileIds = List.of("SOLICITOR_ORG");
+        String orgProfileIds = String.join(",", organisationProfileIds);
+
+        profileRefreshQueueRepository.upsertOrganisationProfileIds(orgProfileIds, 2);
+
+        List<ProfileRefreshQueueEntity> newProfileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(newProfileRefreshQueueEntities, 1, 2);
+    }
+
+    private void validateData(List<ProfileRefreshQueueEntity> profileRefreshQueueEntities,
+                                int expectedSize,
+                                int expectedAccessTypesMinVersion) {
+        assertEquals(expectedSize, profileRefreshQueueEntities.size());
+        profileRefreshQueueEntities.forEach(el -> {
```

Comment:
rename variable el

Reply:
renamed in commit [c2f979a](https://github.com/hmcts/am-org-role-mapping-service/pull/1729/commits/c2f979a5eab2a5c11e7ea4ca33518d905a52e734)